packages/@tailwindcss-upgrade/src/migrate-postcss.ts

```diff
@@ -0,0 +1,95 @@
+import fs from 'node:fs/promises'
+import path from 'node:path'
+import { pkg } from './utils/packages'
+import { info, success, warn } from './utils/renderer'
+
+// Migrates simple PostCSS setups. This is to cover non-dynamic config files
+// similar to the ones we have all over our docs:
+//
+// ```js
+// module.exports = {
+//  plugins: {
+//    tailwindcss: {},
+//    autoprefixer: {},
+//  }
+// }
+export async function migratePostCSSConfig(base: string) {
+  let configPath = await detectConfigPath(base)
+  if (configPath === null) {
+    // TODO: We can look for an eventual config inside package.json
+    return
+  }
+
+  info(`Attempt to upgrade the PostCSS config in file: ${configPath}`)
+
+  let isSimpleConfig = await isSimplePostCSSConfig(base, configPath)
+  if (!isSimpleConfig) {
+    warn(`The PostCSS config contains dynamic JavaScript and can not be automatically migrated.`)
+    return
+  }
+
+  let didAddPostcssClient = false
+  let didRemoveAutoprefixer = false
+
+  let fullPath = path.resolve(base, configPath)
+  let content = await fs.readFile(fullPath, 'utf-8')
+  let lines = content.split('\n')
+  let newLines: string[] = []
+  for (let line of lines) {
+    if (line.includes('tailwindcss:')) {
+      didAddPostcssClient = true
+      newLines.push(line.replace('tailwindcss:', `'@tailwindcss/postcss':`))
```

Comment:
If the previous line here is the nesting plugin, we can remove it as well: 

- https://tailwindcss.com/docs/using-with-preprocessors#nesting

Reply:
also remove `postcss-import` if nothing else is between it and `tailwindcss`